src/commands/init/ai-rules.ts

```diff
@@ -0,0 +1,447 @@
+const decodeHash = (hash: string): string => {
+  try {
+    return atob(hash)
+  } catch (error) {
+    throw new Error(`Failed to decode hash: ${error instanceof Error ? error.message : 'Invalid base64 or URL'}`)
+  }
+}
+
+const fetchProjectInfo = async (url: string): Promise<ProjectInfo> => {
+  try {
+    const response = await fetch(url, {
+      headers: {
+        'content-type': 'text/plain',
+      },
+    })
+
+    if (!response.ok) {
+      throw new Error(`HTTP error! status: ${String(response.status)}`)
+    }
+    const data = (await response.text()) as unknown as string
+    const parsedData = JSON.parse(data) as unknown as ProjectInfo
+    return parsedData
+  } catch (error) {
+    throw new Error(`Failed to fetch project information: ${error instanceof Error ? error.message : 'Unknown error'}`)
+  }
+}
+
+const getRepoUrlFromProjectId = async (api: NetlifyAPI, projectId: string): Promise<string> => {
+  try {
+    const SiteInfo = (await api.getSite({ siteId: projectId })) as SiteInfo
+    const repoUrl = SiteInfo.build_settings?.repo_url
+
+    if (!repoUrl) {
+      throw new Error(`No repository URL found for project ID: ${projectId}`)
+    }
+
+    return repoUrl
+  } catch (error) {
+    if ((error as APIError).status === 404) {
+      throw new Error(`Project with ID ${projectId} not found`)
+    }
+    throw new Error(`Failed to fetch project data: ${(error as Error).message}`)
+  }
+}
+
+const savePrompt = async (instructions: string, ntlContext: string | null, targetDir: string): Promise<void> => {
+  try {
+    const filePath = resolve(targetDir, 'AI-instructions.md')
+    await fs.writeFile(filePath, `Context: ${ntlContext ?? ''}\n\n${instructions}`, 'utf-8')
+    log(`${chalk.green('✅')} AI instructions saved to ${chalk.cyan('AI-instructions.md')}`)
+  } catch (error) {
+    const errorMessage = error instanceof Error ? error.message : 'Unknown error'
+    log(`${chalk.yellow('⚠️')} Warning: Failed to save AI instructions: ${errorMessage}`)
+  }
+}
+
+const cloneRepo = async (repoUrl: string, targetDir: string, debug: boolean): Promise<void> => {
+  try {
+    await runGit(['clone', repoUrl, targetDir], !debug)
+  } catch (error) {
+    throw new Error(`Failed to clone repository: ${error instanceof Error ? error.message : error?.toString() ?? ''}`)
+  }
+}
+
+/**
+ * Handles AI rules initialization workflow
+ * This is the experimental --ai-rules functionality for the init command
+ */
+export const initWithAiRules = async (hash: string, command: BaseCommand): Promise<void> => {
+  // Authenticate first before any API operations
+  await command.authenticate()
+  const { api } = command.netlify
+
+  log(`${chalk.blue('🤖 Initializing AI project')} with rules...`)
+  log(`${chalk.gray('Hash:')} ${hash}`)
+  log(`${chalk.gray('User:')} ${api.accessToken ? 'Authenticated ✅' : 'Not authenticated ❌'}`)
+
+  try {
+    // Step 1: Decode hash and fetch project information
+    log('\n📋 Decoding project hash...')
```

Comment:
I think "decoding project hash" doesn't mean a lot to people. I suggest something more user-friendly.

```suggestion
    log('\n📋 Extracting project details...')
```